lib/ansible/modules/cloud/docker/docker_network.py

```diff
@@ -253,29 +316,43 @@ def has_different_config(self, net):
                     if not (key in net['Options']) or value != net['Options'][key]:
                         different = True
                         differences.append('driver_options.%s' % key)
+
         if self.parameters.ipam_driver:
             if not net.get('IPAM') or net['IPAM']['Driver'] != self.parameters.ipam_driver:
                 different = True
                 differences.append('ipam_driver')
-        if self.parameters.ipam_options:
-            if not net.get('IPAM') or not net['IPAM'].get('Config'):
+
+        if self.parameters.ipam_config is not None and self.parameters.ipam_config:
+            if not net.get('IPAM') or not net['IPAM']['Config']:
                 different = True
-                differences.append('ipam_options')
+                differences.append('ipam_config')
             else:
-                for key, value in self.parameters.ipam_options.items():
-                    camelkey = None
-                    for net_key in net['IPAM']['Config'][0]:
-                        if key == net_key.lower():
-                            camelkey = net_key
-                            break
-                    if not camelkey:
-                        # key not found
-                        different = True
-                        differences.append('ipam_options.%s' % key)
-                    elif net['IPAM']['Config'][0].get(camelkey) != value:
-                        # key has different value
-                        different = True
-                        differences.append('ipam_options.%s' % key)
+                for idx, ipam_config in enumerate(self.parameters.ipam_config):
+                    net_config = dict()
+                    ip_version = get_ip_version(ipam_config['subnet'])
+                    for net_ipam_config in net['IPAM']['Config']:
+                        if ip_version == get_ip_version(net_ipam_config['Subnet']):
+                            net_config = net_ipam_config
+
+                    for key, value in ipam_config.items():
+                        camelkey = None
+                        for net_key in net_config:
+                            if key == net_key.lower():
+                                camelkey = net_key
+                                break
+                        if not camelkey:
```

Comment:
Why not change this to
```suggestion
                        if not camelkey or net_config.get(camelkey) != value:
```
and get rid of the second branch?

Reply:
Sounds good